### DISCUSSION TITLE
Adjustable map/visualization canvas on server side

### DISCUSSION THREAD

**Discussion by merrygin** [mesa/mesa#1602] [Q&A]
I would like to display agents' movements on a world map but only need a zoomed in part of it initially. Ideally, this map part would zoom out if necessary - ie, when an agent steps out of the bounds of the original map part. I thought about using cartopy, but realised that a map is given to the server side once on initialization and won't change afterwards (without restarting the server).

So as far as I understand, this is not possible with mesa out of the box, or is it?

Thanks in advance for any help!

**Comment by tpike3** [comment]
@merrygin sorry for the delay...have you looked at [mesa-geo](https://github.com/projectmesa/mesa-geo) for this?

**Reply by merrygin** [reply]
No worries and thank you! Yes, I asked there as well and while it's not possible yet, they are very interested in adding the feature at some point: https://github.com/projectmesa/mesa-geo/discussions/136
